### ISSUE TITLE
Overriding lodash template options

### ISSUE THREAD

**Issue by tmkasun** [jantimon/html-webpack-plugin#1483]
## Expected behaviour

I expect plugin to pass the lodash template options when building the template file via the loader options. My requirement is , I want to use a handlebar file as the template and output a handlebar file (note: Not a HTML file), So I want to preserve the original handlebar syntax while using different patterns for escape, evaluate, interpolate (i:e `/<#=([\s\S]+?)#>/g`) syntaxes

## Current behaviour

Can't override the loader options

## Environment

Operating system - Mac OS Catalina 10.15.5
Node version - v12.18.1
npm - 6.14.5
webpack@4.28.4 
html-webpack-plugin@4.3.0 

### Config

```js
new HtmlWebpackPlugin({
            inject: false,
            template: 'site/public/pages/index.jag.hbs',
            filename: '../pages/index.jag',
            minify: false,
        }),
```

Copy your template file if it is part of this issue:

```hbs
<html lang="en">

<head>
    <base href="<%= app.context %>/" />
    <meta charset="utf-8" />
    <meta http-equiv="X-UA-Compatible" content="IE=edge" />
    <link rel="shortcut icon" href="<%= app.context %>/site/public/images/favicon.png">
</head>

<body>

    <div id="react-root">
        <div class="apim-dual-ring"></div>
    </div>
    <script src="<%= app.context %>/<#= htmlWebpackPlugin.files.js[0] #>"></script>
    <link rel="stylesheet" href="<%= app.context %>/site/public/fonts/iconfont/material-icons.css">
</body>

</html>
```

## Relevant Links

- N/A

## Additional context

When I debugged the code 

https://github.com/jantimon/html-webpack-plugin/blob/master/lib/loader.js#L9
I found that, If I can pass the template options to above location, Then it would override the default options when invoking the `_.template()` function,

```javascript
const template = _.template(source, _.defaults(options, {interpolate: /<#=([\s\S]+?)#>/g, variable: 'data' }));
```

**Comment by jantimon** [comment]
Probably you will have to use the https://www.npmjs.com/package/underscore-template-loader for all `.hbs` files.

That way the html-webpack-plugin will use that loader and the interpolation options you specified for it.

Theoretically the https://github.com/jantimon/html-webpack-plugin/blob/master/lib/loader.js#L9 would also work however right now it does not allow to provide a string for the interpolate option

**Comment by tmkasun** [comment]
Thank you @jantimon 
That worked perfectly 😃 

I also tried this https://github.com/jantimon/html-webpack-plugin/issues/1220 method , and put a webpack loader config as below,

```javascript
{
                test: /\.jag\.hbs$/,
                loader: 'html-webpack-plugin/lib/loader.js',
                options: {
                    escape: /{%-([\s\S]+?)%}/g,
                    evaluate: /{%([\s\S]+?)%}/g,
                    interpolate: /{%=([\s\S]+?)%}/g,
                },
            },
```

And debugged the code, When I use this `html-webpack-plugin`'s loader it gets triggered 2 times, One with the loader options that i have provided, and one time with no loader options. But when I use `underscore-template-loader` with the same test pattern, `html-webpack-plugin`'s loader did not get triggered.


BTW If someone else having the same problem and looking for a workaround here is my configs

```javascript
module: {
        rules: [
            {
                test: /\.jag\.hbs$/,
                loader: 'underscore-template-loader',
                query: {
                    engine: 'lodash',
                    interpolate: '\\{\\[(.+?)\\]\\}',
                    evaluate: '\\{%([\\s\\S]+?)%\\}',
                    escape: '\\{\\{(.+?)\\}\\}',
                },
            },
    }]
.
.
.
plugins: [
      new HtmlWebpackPlugin({
            inject: false,
            template: 'site/public/pages/index.jag.hbs',
            filename: '../pages/index.jag',
            minify: false,
        }),
    ]
```

Or else you can write a custom loader like below

```javascript
const _ = require('lodash');

module.exports = function (source) {
    /** Used to match template delimiters. */
    const reEscape = /<#-([\s\S]+?)#>/g;
    const reEvaluate = /<#([\s\S]+?)#>/g;
    const reInterpolate = /<#=([\s\S]+?)#>/g;
    const template = _.template(source, {
        variable: 'data', interpolate: reInterpolate, escape: reEscape, evaluate: reEvaluate,
    });
    // Use __non_webpack_require__ to enforce using the native nodejs require
    // during template execution
    return 'var _ = __non_webpack_require__(' + JSON.stringify(require.resolve('lodash')) + ');'
    + 'module.exports = function (templateParams) { with(templateParams) {'
      // Execute the lodash template
      + 'return (' + template.source + ')();'
    + '}}';
};
```

and provide it as the loader in the `HtmlWebpackPlugin` 

```javascript
new HtmlWebpackPlugin({
            inject: false,
            template: '!!/Volumes/resources/admin/source/webpack/hashTemplateLoader.js!site/public/pages/index.jag.hbs',
            filename: '../pages/index.jag',
            minify: false,
        }),
```

**Comment by jantimon** [comment]
You can use `require.resolve('./hashTemplateLoader.js')` to get around the absolute path..

The template loader is executed twice - does it still generate the correct result?

**Comment by tmkasun** [comment]
Yes 

```javascript
allLoadersButThisOne.length > 0 && !force
```

This expression evaluates to false, in both occasions, When there are options and when there are no properties in the options object.

# With options

![image](https://user-images.githubusercontent.com/3313885/87791004-2798e000-c85f-11ea-8c5f-3be308125d76.png)


# Without options

![image](https://user-images.githubusercontent.com/3313885/87790968-1354e300-c85f-11ea-9f13-46cb656332f5.png)


In both cases `!force` is true but `allLoadersButThisOne.length > 0` is false because the length is zero. So it skips the return statement

But in both cases `this.loaders` has two loaders

![image](https://user-images.githubusercontent.com/3313885/87791497-fbca2a00-c85f-11ea-9e5d-12fec01b7a6d.png)

**Comment by jantimon** [comment]
Ah nice so we could use the loader count

**Comment by jantimon** [comment]
Sorry I totally forgot about this issue - pull request in #1623

**Comment by jantimon** [comment]
released in 5.3.0
